Increase size of the "link" field in com_newsfeeds

jeckodevelopment opened the pull request:
Pull Request for Issue #11612.
#### Summary of Changes

This PR increases the size of the `link` field for the com_newsfeeds from 200 to 2048 in order to handle long URLs for Feeds. 

This PR fixes also a mistake in a tip, included in the installation SQL files.
#### Testing Instructions

Apply the patch and try to store a feed URL longer than 200 characters.
#### Documentation Changes Required

Not needed.

jeckodevelopment committed:
increased limit of characters for the URL

To resolve #11612

jeckodevelopment committed:
increased limit of characters for the feed URL

Increased limit of characters for the Feed URL to fix issue #11612

jeckodevelopment committed:
increased limit of characters for the feed URL

To resolve #11612

jeckodevelopment committed:
increased limit of characters for the feed URL

To resolve #11612

jeckodevelopment committed:
increased limit of characters for the feed URL

To fix #11612

andrepereiradasilva commented:
you need to make updates for postegresql and sqlazure too

jeckodevelopment committed:
increased limit of characters for the feed URL

To fix #11612

jeckodevelopment committed:
increased limit of characters for the feed URL

To fix #11612

jeckodevelopment commented:
@andrepereiradasilva can you please check if they are ok?

andrepereiradasilva commented on administrator/components/com_admin/sql/updates/postgresql/3.6.3-2016-08-15.sql:
```diff
@@ -0,0 +1,5 @@
+--- Increasing size of the URL field in com_newsfeeds
+---
+
+ALTER TABLE `#__newsfeeds` ALTER COLUMN `link` TYPE varchar(2048);
```
see https://github.com/jeckodevelopment/joomla-cms/blob/staging/administrator/components/com_admin/sql/updates/postgresql/3.5.0-2016-03-01.sql

andrepereiradasilva commented on administrator/components/com_admin/sql/updates/sqlazure/3.6.3-2016-08-15.sql:
```diff
@@ -0,0 +1,5 @@
+--- Increasing size of the URL field in com_newsfeeds
+---
+
+ALTER TABLE [#__newsfeeds] ALTER COLUMN [link] NVARCHAR(2048) NOT NULL;
```
https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_admin/sql/updates/sqlazure/3.5.0-2016-03-01.sql

andrepereiradasilva commented on administrator/components/com_admin/sql/updates/mysql/3.6.3-2016-08-15.sql:
```diff
@@ -0,0 +1,5 @@
+--- Increasing size of the URL field in com_newsfeeds
+---
+
+ALTER TABLE `#__newsfeeds` MODIFY `link` VARCHAR(2048);
```
dont' we need a  NOT NULL also?

jeckodevelopment committed:
fixed syntax

jeckodevelopment committed:
fixed syntax SQL Azure

jeckodevelopment committed:
fixed syntax

jeckodevelopment commented:
@andrepereiradasilva done. Hope that's ok now :)

jeckodevelopment committed:
Merge branch 'staging' into staging

jeckodevelopment committed:
Merge branch 'staging' into staging

jeckodevelopment committed:
Fixed mistake in comment

Updated comment from
"Set if article is featured" to "Set if contact is featured"

jeckodevelopment committed:
Merge pull request #1 from jeckodevelopment/sqlcommenterror

Fixed mistake in comment

andrepereiradasilva commented:
fix in postgresql too please

jeckodevelopment committed:
Fixed mistake in comment

Fixed, there was "article" instead of "contact"

jeckodevelopment commented:
@andrepereiradasilva done in PostegreSQL, but I can't find it on SQL Azure.
Can you please check/test?

andrepereiradasilva commented on administrator/components/com_admin/sql/updates/mysql/3.6.3-2016-08-15.sql:
```diff
@@ -0,0 +1,5 @@
+---
+--- Increasing size of the URL field in com_newsfeeds
+---
```
use two `--` instead of three `---`
i remember there were problems in the past with that don't know if they are still.
But MySql comments use `--` (http://dev.mysql.com/doc/refman/5.7/en/comments.html) so always better to folow the norm IMO

don't know how it is with the other datababases, but you can check joomla sql files for examples.

jeckodevelopment replied:
I copied from other SQL files, but I'll change of course.

jeckodevelopment committed:
Updated mysql comment to reflect rules

jeckodevelopment committed:
Updated SQL comment to reflect rules

jeckodevelopment committed:
Updated SQL comment to reflect rules

jeckodevelopment commented:
@andrepereiradasilva done for MySQL, Azure SQL, PostgreSQL

andrepereiradasilva commented:
I have tested this item :white_check_mark: successfully on df6f2d1d078f5f504686d5597a631bef2f4bd392<br /><br />on code review

didn't test the upgrade process<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11614">issues.joomla.org/joomla-cms/11614</a>.</sub>

jeckodevelopment committed:
Resolving merge conflicts

to fix merge conflicts after #11617

jeckodevelopment committed:
Resolving merge conflicts

jeckodevelopment committed:
Merge branch 'staging' into staging

jeckodevelopment committed:
Applying patch to increase size of link field

After resolving merge issues, this commit apply again the patch to increase the size of the link field.

andrepereiradasilva commented:
I have tested this item :white_check_mark: successfully on a34afcdb00c15e695d5bb619d0adfdda8c010558<br /><br />just conflicts fixed<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11614">issues.joomla.org/joomla-cms/11614</a>.</sub>

jeckodevelopment commented:
thanks @andrepereiradasilva

jeckodevelopment commented:
@dkanchev can you please test this patch?

zero-24 commented:
I have tested this item :white_check_mark: successfully on a34afcdb00c15e695d5bb619d0adfdda8c010558<br /><br />looks good from here. I have just tryed the mysql part.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11614">issues.joomla.org/joomla-cms/11614</a>.</sub>

zero-24 commented:
@alikon can you double check the non-mysql parts here?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11614">issues.joomla.org/joomla-cms/11614</a>.</sub>

alikon commented:
postgresql  is ok 
sqlazure is ok by review

alikon commented:
I have tested this item :white_check_mark: successfully on a34afcdb00c15e695d5bb619d0adfdda8c010558<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11614">issues.joomla.org/joomla-cms/11614</a>.</sub>

zero-24 commented:
Thanks @alikon

zero-24 commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11614">issues.joomla.org/joomla-cms/11614</a>.</sub>
